tests/std/tests/P0355R7_calendars_and_time_zones_io/test.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <chrono>
+#include <locale>
+#include <sstream>
+
+using namespace std;
+using chrono::duration;
+
+template <class CharT, class Rep, class Period>
+bool test_duration_basic_out(const duration<Rep, Period>& d, const CharT* expected) {
+    basic_ostringstream<CharT> ss;
+
+    ss << d;
+    return ss.str() == expected;
+}
+
+#define WIDEN(TYPE, STR) get<const TYPE*>(make_pair(STR, L##STR));
+
+template <class CharT>
+bool test_duration_locale_out() {
+    basic_stringstream<CharT> ss;
+    const duration<float> d{0.140625};
+    ss.precision(3);
+    ss << d;
+    ss.setf(ios_base::scientific, ios_base::floatfield);
+    ss << ' ' << d;
+
+    basic_string<CharT> expected = WIDEN(CharT, "0.141s 1.406e-01s");
+
+#if !defined(_DLL) || ((_ITERATOR_DEBUG_LEVEL == 0) != defined(_DEBUG))
+    // Using dynamic linking, only release mode supports IDL == 0.

```

Comment:
I believe that this guard is slightly incorrect - it will permit debug IDL=1 when dynamic linking, but that's a non-default setting that might cause mismatch problems.